image/image_test.go

```diff
@@ -0,0 +1,259 @@
+            "/var/log/my-app-logs": {}
+        },
+        "WorkingDir": "/home/alice"
+    },
+    "rootfs": {
+      "diff_ids": [
+        "sha256:c6f988f4874bb0add23a778f753c65efe992244e148a1d2ec2a8b664fb66bbd1",
+        "sha256:5f70bf18a086007016e948b04aed3b82103a36bea41755b6cddfaf10ace3c6ef"
+      ],
+      "type": "layers"
+    },
+    "history": [
+      {
+        "created": "2015-10-31T22:22:54.690851953Z",
+        "created_by": "/bin/sh -c #(nop) ADD file:a3bc1e842b69636f9df5256c49c5374fb4eef1e281fe3f282c65fb853ee171c5 in /"
+      },
+      {
+        "created": "2015-10-31T22:22:55.613815829Z",
+        "created_by": "/bin/sh -c #(nop) CMD [\"sh\"]",
+        "empty_layer": true
+      }
+    ]
+}
+`
+)
+
+var (
+	refStr = `{"digest":"<manifest_digest>","mediaType":"application/vnd.oci.image.manifest.v1+json","size":<manifest_size>}`
+
+	manifestStr = `{
+    "annotations": null,
+    "config": {
+        "digest": "<config_digest>",
+        "mediaType": "application/vnd.oci.image.config.v1+json",
+        "size": <config_size>
+    },
+    "layers": [
+        {
+            "digest": "<layer_digest>",
+            "mediaType": "application/vnd.oci.image.layer.tar+gzip",
+            "size": <layer_size>
+        }
+    ],
+    "mediaType": "application/vnd.oci.image.manifest.v1+json",
+    "schemaVersion": 2
+}
+ `
+)
+
+func TestValidateLayout(t *testing.T) {
+	root, err := ioutil.TempDir("", "oci-test")
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer os.RemoveAll(root)
+
+	err = os.MkdirAll(filepath.Join(root, "blobs", "sha256"), 0700)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = os.MkdirAll(filepath.Join(root, "refs"), 0700)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	desc, err := createLayerFile(root)
+	if err != nil {
+		t.Fatal(err)
+	}
+	manifestStr = strings.Replace(manifestStr, "<layer_digest>", desc.Digest, 1)
+	manifestStr = strings.Replace(manifestStr, "<layer_size>", strconv.FormatInt(desc.Size, 10), 1)
+
+	desc, err = createConfigTestFile(root)
+	if err != nil {
+		t.Fatal(err)
+	}
+	manifestStr = strings.Replace(manifestStr, "<config_digest>", desc.Digest, 1)
+	manifestStr = strings.Replace(manifestStr, "<config_size>", strconv.FormatInt(desc.Size, 10), 1)
+
+	mft, err := createManifestFile(root, manifestStr)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = createRefFile(root, mft)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = createLayoutFile(root)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	err = ValidateLayout(root, []string{"latest"}, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func createLayerFile(root string) (descriptor, error) {
+	layerPath := filepath.Join(root, "blobs", "sha256", "test.tar")
+
+	desc, err := createTarFile(layerPath, []tarContent{
+		tarContent{&tar.Header{Name: "test", Size: 4, Mode: 0600}, []byte("test")},
+	})
+	if err != nil {
+		return descriptor{}, err
+	}
+
+	err = os.Rename(layerPath, filepath.Join(root, "blobs", "sha256", desc.Digest))
```

Comment:
`manifest_test.go` is also doing similar post-creation renaming.  Can we use `createLayerFile` there as well?